StochSS/stochss
# HOTFIX: Spatial Domain Triage

## Testing
- [#1331]
   1. Create a new spatial model and confirm the no error is raised and a message stating the domain has no particles is displayed.
   2. Open the domain section and click `View External Domains`.
   3. Select a domain and confirm that the message is no longer displayed.
   4. Click `Build New Domain`.
   5. In the `Create 3D Domain` section create a new domain.
      - The number of particles for all planes must be non-zero.
      - The diffs of limits for all axes must be > 0.
   6. Click `Build Domain` and confirm that the message is no longer displayed.
- [#1334]
   > This fix requires the SpatialPy v1.0.4 release
   1. Testing is the same as #1335 with the exception of creating and selecting a type.
- [#1335]
   > This issue only occurs with domains that have no particles
   1. Create a new spatial model.
   6. Open the domain section and click `Edit Domain`.
   7. Create a new `Type`.
   8. In the `Create 3D Domain` section create a new domain.
      - The number of particles for all planes must be non-zero.
      - The diffs of limits for all axes must be > 0.
   9. Open `Advanced` and change the type to the type you created.
   10. Click `Build Domain` and confirm that the domain was created successfully.
- [#1336]
   1. Create a new spatial model.
   2. Open the domain section and click `View External Domains`.
   3. Select a domain from the drop down list and click `Import into Model`.
   4. Open the domain section to confirm the selected domain was added successfully.
- [#1342]
   1. Create a new spatial model.
   2. Click `Add Type` and then `Submit`.
   3. Confirm the new type in the `Types` section.

NOTE: The 3D Cylinder domain may have errors due to spaces in the domain type IDs.
